core/DataAccess/Model.php

```diff
@@ -118,13 +129,21 @@ public function deleteArchivesWithPeriod($numericTable, $blobTable, $period, $da
         $query = "DELETE FROM %s WHERE period = ? AND ts_archived < ?";
         $bind  = array($period, $date);
 
-        Db::query(sprintf($query, $numericTable), $bind);
+        $queryObj = Db::query(sprintf($query, $numericTable), $bind);
+        $deletedRows = $queryObj->rowCount();
 
         try {
-            Db::query(sprintf($query, $blobTable), $bind);
+            $queryObj = Db::query(sprintf($query, $blobTable), $bind);
+            $deletedRows += $queryObj->rowCount();
         } catch (Exception $e) {
             // Individual blob tables could be missing
+            $this->logger->debug("Unable to delete archives by period from {blobTable}.", array(
+                'blobTable' => $blobTable,
+                'exception' => $e
+            ));
         }
+
+        return $deletedRows;
```

Comment:
Yes. I wouldn't mock the DB connection. An integration test should be fine and you could check whether it actually returns the deleted count and make sure it only deletes the ones older than the date given